### ISSUE TITLE
老哥 可以介绍一下你bert后接的全连接层吗，新手不是很懂 谢谢

### ISSUE THREAD

**Issue by YiJun-Yang** [1JasonZhang/Q-A-matching-of-real-estate-industry#7]

**Comment by 1JasonZhang** [comment]
`    def forward(self, x, mask, token_type_ids):
        bert_out = self.bert(x, attention_mask = mask, token_type_ids=token_type_ids)
        sentence_emb = bert_out.last_hidden_state[:, 0, :] #取bert最后一层的[cls] token对应的向量作为句子向量
        bert_out = bert_out.last_hidden_state.permute(0, 2, 1).contiguous() # 取最后一层的所有hidden_state shape (batch_size, sequence_length, hidden_size)
        maxpool_out = self.maxpool(bert_out).squeeze(2)# 对最后一层的所有hidden_state做最大池化操作
        avgpool_out = self.avgpool(bert_out).squeeze(2)# 对最后一层的所有hidden_state做均值池化操作
        outputs = torch.cat((maxpool_out, avgpool_out, sentence_emb), 1) # 拼接句子向量， 最大池化向量，均值池化向量
        outputs = self.tanh(self.linear1(outputs)) #进入第一个全连接层用tanh做一下非线性变换
        outputs = self.linear2(outputs) #最后一层全连接输出为（batch_size, num_class）
        return outputs`

**Comment by 1JasonZhang** [comment]
`    def forward(self, x, mask, token_type_ids):
        bert_out = self.bert(x, attention_mask = mask, token_type_ids=token_type_ids)
        sentence_emb = bert_out.last_hidden_state[:, 0, :] #取bert最后一层的[cls] token对应的向量作为句子向量
        bert_out = bert_out.last_hidden_state.permute(0, 2, 1).contiguous() # 取最后一层的所有hidden_state shape (batch_size, sequence_length, hidden_size)
        maxpool_out = self.maxpool(bert_out).squeeze(2)# 对最后一层的所有hidden_state做最大池化操作
        avgpool_out = self.avgpool(bert_out).squeeze(2)# 对最后一层的所有hidden_state做均值池化操作
        outputs = torch.cat((maxpool_out, avgpool_out, sentence_emb), 1) # 拼接句子向量， 最大池化向量，均值池化向量
        outputs = self.tanh(self.linear1(outputs)) #进入第一个全连接层用tanh做一下非线性变换
        outputs = self.linear2(outputs) #最后一层全连接输出为（batch_size, num_class）
        return outputs`

**Comment by 1JasonZhang** [comment]
![image](https://user-images.githubusercontent.com/31090535/100688051-6c5f8680-33bc-11eb-8b9e-257f54a392db.png)

**Comment by YiJun-Yang** [comment]
感谢老哥！

**Comment by YiJun-Yang** [comment]
![image](https://user-images.githubusercontent.com/45503894/100718499-625a7980-33f6-11eb-8e76-d5b640baa43d.png)
hello 请问定义的这个模型是没有用到吗

**Comment by 1JasonZhang** [comment]
这是直接用bert做的句子分类任务，前期用了没有后边的模型效果好，就没用了

**Comment by YiJun-Yang** [comment]
soga(๑•̀ㅂ•́)و✧

**Comment by zobwink** [comment]
你好 请问forward里这一系列池化操作是根据什么原理，有什么作用啊

**Comment by 1JasonZhang** [comment]
最后一层所有的输出是一个3维的张量，用池化操作可以降维到2维，可以提取关键特征，然后与句子向量拼接共同组成特征向量输入到全连接中。
